### PR TITLE
WIP: Cockpit

### PR DESCRIPTION
**Pull request by vignanl**
Addition of cockpit module to main menu is done.  Pending tasks are

- actions.py  file for cockpit module
- Complete testing of whether cockpit module is available under /cockpit url

### CODE REVIEW

**Comment by JosephKiranBabu** [comment]
Feature story: https://github.com/freedombox/Plinth/issues/838

**Comment by SunilMohanAdapa** [comment]
To give more direction on this work, I explored a bit and I was able to get cockpit working behind Apache.

TODO:

1. Cockpit application should ship following Apache configuration:
   ```
   ##
   ## On all sites, provide Cockpit on a default path: /cockpit
   ##
   ## Requires the following Apache modules to be enabled:
   ##   mod_headers
   ##   mod_proxy
   ##   mod_proxy_http
   ##   mod_proxy_wstunnel
   ##
   <Location /cockpit>
       ProxyPass        http://localhost:9090/
   </Location>

   <Location /cockpit/>
       ProxyPass        http://localhost:9090/cockpit/
   </Location>

   <Location /cockpit/socket>
       ProxyPass        ws://localhost:9090/cockpit/socket
   </Location>
   ```
   The URL to use is `http://<freedombox>/cockpit` (without a trailing slash). This can perhaps be improved to work with and without the trailing slash.
1. In the setup action of cockpit we must enable Apache module `proxy_wstunnel` and restart Apache server.
1. It is necessary to change cockpit configuration to allow access from outside.  Edit `/etc/cockpit/cockpit.conf`
   ```
   [WebService]
   Origins = https://localhost:9090 https://localhost:4430
   ```
   The `Origin:` header must be checked by a server using WebSockets.  This is the only way to avoid cross origin requests since same-origin policy (and CORS) don't apply to WebSockets.  This means the Plinth's cockpit application must listen to the domain added/removed/changed signals and modify this configuration file accordingly. When a domain is added, it should be added to the Origins list.  When a domain is removed, it should be removed from the Origins list. When a domain is changed, it should be changed in the Origins list. It is a grave security error if the Origins list above is empty.

File: data/etc/apache2/conf-available/cockpit-plinth.conf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+## On all sites, provide cockpit on a defaut path: /cockpit
+##
+## Requires the following Apache modules to be enabled:
+##   mod_headers
+##   mod_proxy
+##   mod_proxy_http
+##   mod_proxy_wstunnel
+##
+
+<Location /cokcpit>
```
Comment (JosephKiranBabu) [inline data/etc/apache2/conf-available/cockpit-plinth.conf]: Typo in the word cockpit

Comment (vignanl) [inline data/etc/apache2/conf-available/cockpit-plinth.conf]: Thanks

**Comment by JosephKiranBabu** [comment]
@vignanl Is this feature ready for review? If so, please remove the WIP tag in the title.

**Comment by JosephKiranBabu** [comment]
@vignanl I pulled your branch and did some testing. Observed the following things.
1. The actions file must be renamed from `cockpit.socket` to `cockpit`. Plinth was unable to find it since the framework expects the name to `cockpit`.
2. I was unable to login to cockpit using my Plinth admin account.

**Comment by JosephKiranBabu** [comment]
@SunilMohanAdapa I think cockpit belongs in the /sys page and not under /apps since it's a system administration tool. Do you think otherwise?
Maybe we can also delegate some of the menu items in the Configuration page to cockpit, so it makes sense for it to be on that page.

**Comment by SunilMohanAdapa** [comment]
I have continued work on this in #1116.  All the comments have been addressed there. Closing this one.

**Pull request closed without merging**